module/platform/webots/Webots/src/Webots.cpp

```diff
@@ -280,67 +272,71 @@ namespace module::platform::webots {
+                .then([this](const ServoTargets& commands, const DarwinSensors& sensors) {
+                    // ************************** Receiving ***************************************
+                    // Get the size of the message
+                    uint32_t Nn;
+                    if (recv(fd, &Nn, sizeof(Nn), 0 != sizeof(Nn))) {
+                        log<NUClear::ERROR>("Failed to read message size from TCP connection");
+                        return;
+                    }
+
+                    // Convert from network endian to host endian
+                    const uint32_t Nh = ntohl(Nn);
+
+                    // Get the message
+                    std::vector<char> data(Nh, 0);
+                    if (uint64_t(recv(fd, data.data(), Nh, 0)) != Nh) {
+                        log<NUClear::ERROR>("Failed to read message from TCP connection");
+                        return;
+                    }
+
+                    // Deserialise the message into a neutron
+                    SensorMeasurements msg = NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(data);
+
+                    translate_and_emit_sensor(msg);
+
+                    // Service the watchdog
+                    emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
+
+                    // ****************************** TIME **************************************
+
+                    // Deal with time
+
+                    // Save our previous deltas
+                    const uint32_t prev_sim_delta  = sim_delta;
+                    const uint32_t prev_real_delta = real_delta;
+
+                    // Update our current deltas
+                    sim_delta  = msg.time - current_sim_time;
+                    real_delta = msg.real_time - current_real_time;
+
+                    // Calculate our custom rtf - the ratio of the past two sim deltas and the past two real time deltas
+                    utility::clock::custom_rtf = static_cast<double>(sim_delta + prev_sim_delta)
+                                                 / static_cast<double>(real_delta + prev_real_delta);
+
+                    // Update our current times
+                    current_sim_time  = msg.time;
+                    current_real_time = msg.real_time;
+
+                    ActuatorRequests to_send_now = make_acutator_request(commands, sensors);
+
+                    // ***************** Sending ********************************
+                    data = NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(to_send_now);
+                    // Size of the message, in network endian
+                    Nn = htonl(data.size());
+                    // Send the message size first
+                    if (send(fd, &Nn, sizeof(Nn), 0) == sizeof(Nn)) {
+                        log<NUClear::ERROR>(
+                            fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
```

Comment:
`errno` isn't what you want it to be here

Reply:
:question:

Reply:
https://man.archlinux.org/man/send.2.en#RETURN_VALUE says otherwise.

Reply:
Oh shit. I forgot that shit was a meme global. mb